### PR TITLE
modified ast

### PR DESCRIPTION
**Pull request by ZaidQureshi**
this is the ast I am thinking. As per Elsa's suggestions I removed strings and ints. The Term type represents both atoms and compound terms.

### CODE REVIEW

File: src/common.ml [hunk 1]
```diff
@@ -20,17 +20,16 @@ type token =
 
 (* Constants *)
 type const =
-      AtomConst of string    (* atoms    *)
-    | IntConst of int        (* integers *)
-    | FloatConst of float    (* floats   *)
-    | StringConst of string  (* strings  *)
```
Comment (adamjstewart) [inline src/common.ml]: Let's keep `IntConst`, `FloatConst`, and `StringConst` for now. You don't need to handle them in your evaluator yet, but I don't want to dumb down my lexer if I've already done the work.

File: src/common.ml [hunk 2]
```diff
@@ -20,17 +20,16 @@ type token =
...
+   | BoolConst of bool              (* booleans *)
 
 (* Expressions *)
 type exp =
-      VarExp of string              (* variables                *)
-    | ConstExp of const             (* constants                *)
-    | RuleExp of exp * exp          (* Head :- Body.            *)
-    | FactExp of exp                (* Head.                    *)
-    | CompoundTermExp of exp * exp  (* functor(arg1, arg2, ...) *)
-    | ListExp of exp * exp          (* arg1, arg2               *)
```
Comment (adamjstewart) [inline src/common.ml:36]: I'm not sure if I understand any of the changes you made to `exp`.  Can you explain them? It seems like you moved most of the types of expressions to constants. You also currently have no way of representing a rule with conjunctions with this new AST.

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Did you change the indentation to 3 spaces? Can you change it back to 4?

**Comment by ZaidQureshi** [comment]
ok i changed the name const to atomic. I put Terms (both compound terms and atoms) in there because to me they seem like an atomic thing just like a boolean const. I added the pred type to allow for a signle term, or a conjunction of preds or a disjunction of preds. Head is just an atomic.

**Pull request closed without merging**